source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -200,7 +200,7 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
       // Currently it is not possible to add {{"append": "1"}, {"append": "2"}} (the intended
       // combined headers: {{"original": "true"}, {"append": "1"}, {"append": "2"}}) to the request
       // to upstream server by only sets `headers_to_append`.
-      if (header_to_modify != nullptr) {
+      if (!header_to_modify.empty()) {
```

Comment:
In general I wonder if it's worth a docs comment somewhere that header to append appends to first value.
If had a tracing header in a non-envoy server which did multi-value,
Trace: ServerA
Trace: ServerB
I think Envoy as server C would generally end up
Trace: ServerA, ServerC
Trace: ServerB.
We offer no header order guarantees so really anyone who cares about order is ideally doing inline appends anyway, so optional.

Reply:
Honestly I find this code mind boggingly confusing every time I look at this, between gRPC/HTTP/etc. @dio do you mind taking a look and seeing if we can improve the docs here for extauth about this behavior?